app/models/recent_activity.rb

```diff
@@ -69,6 +69,12 @@ def self.user_activity(user, reject_keys=[])
   # All activities that are public and should be visible for a user
   # * +user+ - the user which activities will be returned
   def self.user_public_activity user
```

Comment:
I think we can use an SQL here instead of a `map`+SQL. Try:

```ruby
user_activity(user, ["space.decline"])
  .where("activities.parameters REGEXP ':user_id: ?[^0-9]'", user.id)
```

REGEXP is slow on SQLs but still should be a lot faster than a `map` plus another query.